pkg/generate/app/cmd/newapp.go

```diff
@@ -662,7 +662,7 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 			switch {
 			case refInput.ExpectToBuild:
 				glog.V(4).Infof("will add %q secrets into a build for a source build of %q", strings.Join(c.Secrets, ","), refInput.Uses)
-				if err := refInput.Uses.AddBuildSecrets(c.Secrets); err != nil {
+				if err := refInput.Uses.AddBuildSecrets(c.Strategy, c.Secrets); err != nil {
```

Comment:
@bparees I was afraid that somebody realizes that and kinda hoping @csrwng will tell me that field is always set and it is set after the detection of Dockerfile in source repo happens :-) Otherwise I will have to investigate how to inject secrets after we have that information.